CsMerge/MergeHelper.cs

```diff
@@ -95,7 +107,7 @@ public static class MergeHelper<T> where T: class, IKeyedEntry {
                let m = GetValue( localObj, id )
                let t = GetValue( theirObj, id )
                let mergeResult = Merge( b, m, t, delModResolver, contentResolver )
-               orderby mergeResult.ResolvedItem.Key
+               orderby mergeResult.Key
```

Comment:
This was causing a null reference exception before as the ResolvedItem can be null (for example if deleted or if it isn't relevant). I have changed so the key is stored separately.